### PR TITLE
Add handles search post endpoint

### PR DESCRIPTION
**Pull request by bigirishlion**

### CODE REVIEW

File: src/routes/handles.route.ts [hunk 1]
```diff
@@ -12,27 +12,12 @@ class HandlesRoute extends BaseRoute {
 
     private initializeRoutes() {
         this.router.get(`${this.path}`, this.injectRegistryMiddleware, this.handlesController.getAll);
+        this.router.post(`${this.path}/search`, this.injectRegistryMiddleware, this.handlesController.search);
```
Comment (papag00se) [inline src/routes/handles.route.ts:15]: After thinking on this, I think `/list` is appropriate for this endpoint. I can explain more when we chat, but basically, you can either start from the full list of Handles (`/handles`) or you can start from a predefined list of Handles (`/handles/list`) and from there all of the current filter params work from the query string.

**Review by papag00se — CHANGES_REQUESTED** [review]

File: src/repositories/memory/handles.repository.ts [hunk 2]
```diff
@@ -62,14 +62,16 @@ class MemoryHandlesRepository implements IHandlesRepository {
...
                       }
                       return agg;
                   }, [])
                 : HandleStore.getHandles().reduce<StoredHandle[]>((agg, handle) => {
-                      if (!search || (search && (handle.name.includes(search) || handle.hex.includes(search)))) {
-                          agg.push(handle);
-                      }
+                      if (search && !(handle.name.includes(search) || handle.hex.includes(search))) return agg;
+                      if (handles && !(handles.includes(handle.name) || handles.includes(handle.hex))) return agg;
+
```
Comment (papag00se) [inline src/repositories/memory/handles.repository.ts:73]: I think these two can combine.

Also, I think doing the hex search this way could break for Handles that ARE hex

File: src/controllers/handles.controller.ts [hunk 3]
```diff
@@ -57,6 +57,49 @@ class HandlesController {
...
+
+            const search = new HandleSearchModel({
+                characters,
+                length,
+                rarity,
+                numeric_modifiers,
+                search: searchQuery,
+                holder_address,
+                personalized,
+                og
```
[the remark this answers is not in the document]
Comment (bigirishlion) [reply src/controllers/handles.controller.ts:72]: Good catch!

File: src/controllers/handles.controller.ts [hunk 4]
```diff
@@ -57,6 +57,49 @@ class HandlesController {
...
+                const handles = await handleRepo.getAllHandleNames(search, sortParam);
+                res.set('Content-Type', 'text/plain; charset=utf-8');
+                res.set('x-handles-search-total', handles.length.toString());
+                res.status(handleRepo.currentHttpStatus()).send(handles.join('\n'));
+                return;
+            }
+
+            let result = await handleRepo.getAll({ pagination, search });
+
+            res.set('x-handles-search-total', result.searchTotal.toString())
```
[the remark this answers is not in the document]
Comment (bigirishlion) [reply src/controllers/handles.controller.ts:95]: Yup. I'll fix it

**Pull request merged**